http2_test.go:52: Expected HTTP/2 connection to server, but connection was using HTTP/1.1

The test suite fails here using Go 1.7 on Debian unstable amd64:
```
export GOPATH=$(mktemp -d)
go get -v github.com/tylerb/graceful
go get -v golang.org/x/net/http2
```
```
go test -v github.com/tylerb/graceful
=== RUN   TestGracefulRun
--- PASS: TestGracefulRun (0.25s)
=== RUN   TestGracefulRunLimitKeepAliveListener
--- PASS: TestGracefulRunLimitKeepAliveListener (0.25s)
=== RUN   TestGracefulRunTimesOut
--- PASS: TestGracefulRunTimesOut (0.60s)
=== RUN   TestGracefulRunDoesntTimeOut
--- PASS: TestGracefulRunDoesntTimeOut (1.00s)
=== RUN   TestGracefulRunDoesntTimeOutAfterConnectionCreated
--- PASS: TestGracefulRunDoesntTimeOutAfterConnectionCreated (1.10s)
=== RUN   TestGracefulRunNoRequests
--- PASS: TestGracefulRunNoRequests (0.00s)
=== RUN   TestGracefulForwardsConnState
--- PASS: TestGracefulForwardsConnState (0.25s)
=== RUN   TestGracefulExplicitStop
--- PASS: TestGracefulExplicitStop (0.10s)
=== RUN   TestGracefulExplicitStopOverride
--- PASS: TestGracefulExplicitStopOverride (0.10s)
=== RUN   TestBeforeShutdownAndShutdownInitiatedCallbacks
--- PASS: TestBeforeShutdownAndShutdownInitiatedCallbacks (0.10s)
=== RUN   TestBeforeShutdownCanceled
--- PASS: TestBeforeShutdownCanceled (0.60s)
=== RUN   TestNotifyClosed
--- PASS: TestNotifyClosed (0.01s)
=== RUN   TestStopDeadlock
--- PASS: TestStopDeadlock (0.10s)
=== RUN   TestStopRace
--- PASS: TestStopRace (0.00s)
=== RUN   TestInterruptLog
--- PASS: TestInterruptLog (0.00s)
=== RUN   TestMultiInterrupts
--- PASS: TestMultiInterrupts (0.00s)
=== RUN   TestLogFunc
--- PASS: TestLogFunc (0.00s)
=== RUN   TestHTTP2ListenAndServeTLS
--- FAIL: TestHTTP2ListenAndServeTLS (0.14s)
	http2_test.go:52: Expected HTTP/2 connection to server, but connection was using HTTP/1.1
=== RUN   TestHTTP2ListenAndServeTLSConfig
--- FAIL: TestHTTP2ListenAndServeTLSConfig (0.11s)
	http2_test.go:52: Expected HTTP/2 connection to server, but connection was using HTTP/1.1
FAIL
exit status 1
FAIL	github.com/tylerb/graceful	4.730s
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,7 @@
 language: go
 sudo: false
 go:
+  - 1.7
   - 1.6.2
   - 1.5.4
   - 1.4.3

--- a/graceful.go
+++ b/graceful.go
@@ -180,6 +180,9 @@ func (srv *Server) ListenTLS(certFile, keyFile string) (net.Listener, error) {
 		return nil, err
 	}
 
+	// Enable http2
+	enableHTTP2ForTLSConfig(config)
+
 	conn, err := srv.newTCPListener(addr)
 	if err != nil {
 		return nil, err
@@ -189,6 +192,28 @@ func (srv *Server) ListenTLS(certFile, keyFile string) (net.Listener, error) {
 
 	tlsListener := tls.NewListener(conn, config)
 	return tlsListener, nil
+}
+
+// Enable HTTP2ForTLSConfig explicitly enables http/2 for a TLS Config. This is due to changes in Go 1.7 where
+// http servers are no longer automatically configured to enable http/2 if the server's TLSConfig is set.
+// See https://github.com/golang/go/issues/15908
+func enableHTTP2ForTLSConfig(t *tls.Config) {
+
+	if TLSConfigHasHTTP2Enabled(t) {
+		return
+	}
+
+	t.NextProtos = append(t.NextProtos, "h2")
+}
+
+// TLSConfigHasHTTP2Enabled checks to see if a given TLS Config has http2 enabled.
+func TLSConfigHasHTTP2Enabled(t *tls.Config) bool {
+	for _, value := range t.NextProtos {
+		if value == "h2" {
+			return true
+		}
+	}
+	return false
 }
 
 // ListenAndServeTLS is equivalent to http.Server.ListenAndServeTLS with graceful shutdown enabled.

--- a/http2_test.go
+++ b/http2_test.go
@@ -48,7 +48,7 @@ func checkIfConnectionToServerIsHTTP2(t *testing.T, wg *sync.WaitGroup, c chan o
 		t.Fatalf("Error encountered while connecting to test server: %s", err)
 	}
 
-	if r.Proto != "HTTP/2.0" {
+	if !r.ProtoAtLeast(2, 0) {
 		t.Fatalf("Expected HTTP/2 connection to server, but connection was using %s", r.Proto)
 	}
 }
@@ -108,6 +108,7 @@ func TestHTTP2ListenAndServeTLSConfig(t *testing.T) {
 
 		tlsConf := &tls.Config{
 			Certificates: []tls.Certificate{cert},
+			NextProtos:   []string{"h2"}, // We need to explicitly enable http/2 in Go 1.7+
 		}
 
 		tlsConf.BuildNameToCertificate()

